htdocs/adherents/class/adherentstats.class.php

```diff
@@ -181,4 +182,56 @@ public function getAllByYear()
 
 		return $this->_getAllByYear($sql);
 	}
+
+
+	/**
+	 *	Return count of member by status group by adh type, total and average
+	 *
+	 * 	@return		array                   Array with
+	 */
+	public function countMembersByTypeAndStatus()
+	{
+		global $user;
+
+		$now = dol_now();
+
+		$sql = "SELECT COALESCE(d.fk_adherent_type, 'total') as fk_adherent_type, t.libelle as label";
+		$sql .= ", COUNT(CASE WHEN d.statut = ".Adherent::STATUS_DRAFT." THEN 'members_draft' ELSE NULL END) as members_draft";
```

Comment:
Do not use CASE WHEN into sql, use instead  $db->ifsql method that will generate the sql using the if with the syntax according to database manager.

Reply:
Done